es6.js

```diff
@@ -7,35 +7,33 @@ module.exports = {
   plugins: ['@babel', 'html'],
   extends: ['airbnb-base', 'plugin:import/typescript'],
   env: {
-    es6: true,
+    es2021: true,
   },
   parser: '@babel/eslint-parser',
   parserOptions: {
-    ecmaVersion: 2020,
+    ecmaVersion: 'latest',
     sourceType: 'module',
     requireConfigFile: false,
   },
-  settings: {
-    'import/extensions': ['.js', '.json', '.ts', '.tsx'],
-  },
   rules: {
     'no-console': ['warn', { allow: ['error', 'trace', 'time'] }],
     'no-param-reassign': 'off',
     'no-plusplus': 'off',
-    'no-unused-expressions': 'off', // Disable for better @babel version
-    '@babel/no-unused-expressions': ['warn', { allowShortCircuit: true }],
     'no-use-before-define': ['error', { functions: false, classes: true, variables: true }],
-    'no-warning-comments': 'warn',
+    'no-warning-comments': ['warn', { terms: ['FIXME', 'TODO', 'HACK'], location: 'anywhere' }],
     'prefer-destructuring': 'off',
     camelcase: 'off',
-
+    'no-unused-expressions': 'off', // Disable for better @babel version
+    '@babel/no-unused-expressions': ['warn', { allowShortCircuit: true }],
+    'import/extensions': ['error', 'ignorePackages', { js: 'never', jsx: 'never', ts: 'never', tsx: 'never' }],
     'import/no-extraneous-dependencies': [
       'error',
       {
         devDependencies: [
-          '**/*.stories.js',
-          '**/*.test.js',
-          '**/*.spec.js',
+          '**/setupTests.[tj]s?(x)',
```

Comment:
Note from Jordi: 
Allow import/no-extraneous-dependencies to look at setupTests file for devDeps. Also supports typescript files by default

Reply:
I'd like more context for this one.

Reply:
no-extraneous-dependencies looks at your package dependencies to make sure you have a dependency that you are using in your project. if the file is not matched in this devDependencies list, the dependency needs to be in your nonDevDeps. If the file is matched to one of these matchers in devDependencies, then it checks the devDependencies list in your package.json and makes sure its there. Of course if a devDep is in the package.json's dependencies instead of devDependencies, the rule won't rightly complain.

I just updated that test, stories, and spec files can either be JS or TS files instead of only JS files. Then I added setupTests file to be in here because any imports in there will be a devDep since that file is run before tests to set them up.